geary-papermc-tracking/src/main/kotlin/com/mineinabyss/geary/papermc/tracking/blocks/systems/TrackOnSetBlockComponent.kt

```diff
@@ -0,0 +1,30 @@
+package com.mineinabyss.geary.papermc.tracking.blocks.systems
+
+import com.mineinabyss.geary.annotations.Handler
+import com.mineinabyss.geary.papermc.tracking.blocks.components.SetBlock
+import com.mineinabyss.geary.papermc.tracking.blocks.gearyBlocks
+import com.mineinabyss.geary.prefabs.PrefabKey
+import com.mineinabyss.geary.systems.GearyListener
+import com.mineinabyss.geary.systems.accessors.TargetScope
+import org.bukkit.block.data.type.*
+
+// Adds
+class TrackOnSetBlockComponent : GearyListener() {
+    private val TargetScope.block by get<SetBlock>()
+    private val TargetScope.prefab by onSet<PrefabKey>()
+
+    @Handler
+    fun TargetScope.loadBlock() {
+        val blockData = gearyBlocks.blockMap.entries.firstOrNull {
+            it.value == block.blockId &&
```

Comment:
This is working a bit backwards, maybe you want a map the other way around, i.e. `(block type + id) -> blockstate`. You could use Pair for it, or technically better might be a map of `block type -> array of blockstates`, where the index is our id (I actually like this structure because it makes it really clear ids can repeat, there aren't any gaps, and they're unique to the block type)